language/move-lang/src/hlir/translate.rs

```diff
@@ -399,7 +399,12 @@ fn type_name(_context: &Context, sp!(loc, ntn_): N::TypeName) -> H::TypeName {
     use H::TypeName_ as HT;
     use N::TypeName_ as NT;
     let tn_ = match ntn_ {
-        NT::Multiple(_) => panic!("ICE type constraints failed {}:{}", loc.file(), loc.span()),
+        NT::Multiple(_) => panic!(
+            "ICE type constraints failed {}:{}-{}",
+            loc.file(),
+            loc.start(),
+            loc.end()
+        ),
```

Comment:
Related to my above comment: changes like this wouldn't be necessary if a `std::ops::Range` were used, since like `codespan::Span`, it implements the `Display` trait.

Reply:
I'd rather not just because of the uncertainty right now around interning of the filename. We can always add it later. 

Besides, I want to actually switch all of these panics to a `diag!()` anyway, which will make it sort of moot